rosidl_typesupport_opendds_cpp/resource/srv__type_support.cpp.em

```diff
@@ -104,15 +111,119 @@ void * create_requester__@(service.namespaced_type.name)(
     DDS::Subscriber_var dds_subscriber,
     allocator_t allocator)
 {
-@# TODO: Implement, considering original code in ffe10f9 or earlier
-    return nullptr;
+  using RequesterType = rosidl_typesupport_opendds_cpp::Requester<
+    @(__dds_request_msg_type),
+    @(__dds_response_msg_type)
+  >;
+
+  @# Register TypeSupports
+  @(__dds_msg_typesupport_type)RequestWrapperTypeSupport_var tsRequest =
+    new @(__dds_msg_typesupport_type)RequestWrapperTypeSupportImpl;
+
+  if (tsRequest->register_type(dds_participant, "") != DDS::RETCODE_OK) {
+    RMW_SET_ERROR_MSG("C++ exception during registering of request message type");
+    return NULL;
+  }
+
+  @(__dds_msg_typesupport_type)ResponseWrapperTypeSupport_var tsResponse =
+    new @(__dds_msg_typesupport_type)ResponseWrapperTypeSupportImpl;
+
+  if (tsResponse->register_type(dds_participant, "") != DDS::RETCODE_OK) {
+    RMW_SET_ERROR_MSG("C++ exception during registering of response message type");
+    return NULL;
+  }
+
+  @# Create Topics
+  CORBA::String_var type_name = tsRequest->get_type_name();
+  DDS::Topic_var request_topic = dds_participant->create_topic(request_topic_str,
+                                                      type_name.in(),
+                                                      TOPIC_QOS_DEFAULT,
+                                                      DDS::TopicListener::_nil(),
+                                                      OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+  if (CORBA::is_nil(request_topic.in())) {
+    RMW_SET_ERROR_MSG("Request create_topic failed");
+    return NULL;
+  }
+
+  type_name = tsResponse->get_type_name();
+  DDS::Topic_var response_topic = dds_participant->create_topic(response_topic_str,
+                                        type_name.in(),
+                                        TOPIC_QOS_DEFAULT,
+                                        DDS::TopicListener::_nil(),
+                                        OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+  if (CORBA::is_nil(response_topic.in())) {
+    RMW_SET_ERROR_MSG("Response create_topic failed");
+    return NULL;
+  }
+
+  @# Get QoS
+  DDS::DataWriterQos dw_qos;
+  dds_publisher->get_default_datawriter_qos(dw_qos);
+
+  DDS::DataReaderQos dr_qos;
+  dds_subscriber->get_default_datareader_qos(dr_qos);
```

Comment:
This is the most flexible operation but I'm wondering if the QoS won't always be the same for the two?

Reply:
This logic was not needed - removed.